### ISSUE TITLE
Slow types error when library uses .d.mts

### ISSUE THREAD

**Issue by erictaylor** [jsr-io/jsr#346]
I get the following error when doing a dry run publish where a dependency library's included `.d.mts` file seems not to be recognized?

```sh
error[unsupported-nested-javascript]: referenced a JavaScript module without type declarations from a TypeScript module
 --> /[redacted]/clsx@2.1.0/node_modules/clsx/clsx.d.mts
  = hint: add a type declaration (d.ts) for the JavaScript module, or rewrite it to TypeScript

  info: JavaScript files with no corresponding declaration require type inference to be type checked
  info: fast check avoids type inference, so referencing a JavaScript file with no type declarations is not supported
  docs: https://jsr.io/go/slow-type-unsupported-nested-javascript

This package contains errors for slow types. Fixing these errors will:

  1. Significantly improve your package users' type checking performance.
  2. Improve the automatic documentation generation.
  3. Enable automatic .d.ts generation for Node.js.

Don't want to bother? You can choose to skip this step by
providing the --allow-slow-types flag.

error: Found 1 problem
```

Package in question is [clsx](https://github.com/lukeed/clsx).

My module imports and consumes the following:

```ts
import { type ClassValue, clsx } from 'clsx';

// ... some exported types include references to `ClassValue`
```

The libraries `package.json` (omitted keys not relevant):

```jsonc
{
  "name": "clsx",
  "version": "2.1.0",
  "repository": "lukeed/clsx",
  "description": "A tiny (239B) utility for constructing className strings conditionally.",
  "module": "dist/clsx.mjs",
  "unpkg": "dist/clsx.min.js",
  "main": "dist/clsx.js",
  "types": "clsx.d.ts",
  "license": "MIT",
  "exports": {
    ".": {
      "import": {
        "types": "./clsx.d.mts",
        "default": "./dist/clsx.mjs"
      },
      "default": {
        "types": "./clsx.d.ts",
        "default": "./dist/clsx.js"
      }
    },
    "./lite": {
      "import": {
        "types": "./clsx.d.mts",
        "default": "./dist/lite.mjs"
      },
      "default": {
        "types": "./clsx.d.ts",
        "default": "./dist/lite.js"
      }
    }
  },
  "files": [
    "*.d.mts",
    "*.d.ts",
    "dist"
  ]
}
```

**Comment by lucacasonato** [comment]
This may have been fixed by #425?

@dsherret WDYT?

**Comment by dsherret** [comment]
This will be fixed by https://github.com/denoland/deno/pull/23173
